index.js

```diff
@@ -29,67 +28,64 @@ DatabaseUpdates.prototype = Object.create(EventEmitter.prototype)
 
 DatabaseUpdates.prototype.getFiles = function getFiles() {
   try {
-    this.updateFiles = fs.readdirSync(this.updatePath)
+    this.updateFiles = fs
+      .readdirSync(this.updatePath)
       // exclude non-javascript files in the updates folder
-      .map(i => (path.extname(i) !== '.js' ? false : i))
+      .map((i) => (path.extname(i) !== '.js' ? false : i))
       // exclude falsy values and filenames that without a valid semver
-      .filter(i => i && semver.valid(i.split('-')[0]))
+      .filter((i) => i && semver.valid(i.split('-')[0]))
       // exclude anything after a hyphen from the version number
       .sort((a, b) => semver.compare(a.split('-')[0], b.split('-')[0]))
   } catch (e) {
     this.logger.error('Error reading updatePath folder', e)
   }
 }
 
-DatabaseUpdates.prototype.updateExists = function updateExists(file, cb) {
-  this.updateCollection.count({ file }, (err, count) => {
-    if (err) return cb(err)
-    return cb(null, count !== 0)
-  })
+DatabaseUpdates.prototype.updateExists = function updateExists(file) {
+  return this.updateCollection.count({ file }).then((count) => count !== 0)
 }
 
-DatabaseUpdates.prototype.runFile = function runFile(file, next) {
-  this.updateExists(file, (err, exists) => {
-    if (err) return next(err)
-    if (exists) return next()
-
-    this.emit('file', file)
-    this.logger.info('Running update:', file)
+DatabaseUpdates.prototype.runFile = async function runFile(file) {
+  const exists = await this.updateExists(file)
+  if (exists) return Promise.resolve()
 
-    /* eslint-disable global-require, import/no-dynamic-require */
-    const update = require(path.join(this.updatePath, file))
-    /* eslint-enable global-require, import/no-dynamic-require */
+  this.emit('file', file)
+  this.logger.info('Running update:', file)
 
-    return update(this.db, (err) => {
-      if (err) {
-        this.logger.error('Error running update:', file)
-        return next(err)
-      }
+  /* eslint-disable global-require, import/no-dynamic-require */
+  const update = require(path.join(this.updatePath, file))
+  /* eslint-enable global-require, import/no-dynamic-require */
 
-      return this.persistUpdate(file, next)
+  return update(this.db)
+    .then(() => this.persistUpdate(file))
+    .catch((err) => {
+      this.logger.error('Error running update:', file)
```

Comment:
I follow the pattern, but curious why not emit an 'error'?

Reply:
no rollback/revert? I don't know about them, but MongoDB transactions are a thing, right?

say the creation of the migration worked, but running it failed. unless I'm misunderstanding the code, I don't see that failure persisted to the DB (or the DB document deleted) - so the library can attempt to re-run it or skip it in the future.

Reply:
Yeah I didnt build in rollbacks or reverts when I first wrote this.